validation.go

```diff
@@ -149,3 +192,79 @@ func convertValidations(validations []Validation) []cpluginv1.ParameterValidatio
 	}
 	return out
 }
+
+// ApplyConfigValidations a utility function that applies all the validations for parameters, return an error that
+// consists of a combination of errors from the configurations.
+func applyConfigValidations(params map[string]Parameter, config map[string]string) error {
```

Comment:
Last comment! 😄 I propose grouping these validation-related functions under a private struct, this way we wouldn't pollute the package with static functions. The type could even be a `map[string]Parameter`, then all functions will have access to the parameters already.
```
type validator map[string]Parameter

func (v validator) Validate(config map[string]string) error {
    ...
}

// we can use it like this
err := validator(source.Params()).Validate(config)
```

This is just one way to do it, but in any case I think creating a struct for this would make the code a bit nicer 😉